### PR TITLE
Added ZMQ protocol service to Zarb

### PR DESCRIPTION
**Pull request by nagarajmanjunath**
## Description

* Added ZMQ protocol service to Zarb.
* update the node service by adding zmq start server method.
* update the main config by adding zmq config.
* update the config test cases.

## Checklist

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] CHANGELOG is updated.

### CODE REVIEW

File: cmd/daemon/start.go [hunk 1]
```diff
@@ -186,10 +186,10 @@ func getValidatorKeyFromWallet(walletPath string) (string, error) {
 	if len(addrInfos) == 0 {
 		return "", fmt.Errorf("validator address is not defined")
 	}
-	password := ""
-	if wallet.IsEncrypted() {
-		password = cmd.PromptPassword("Wallet password", false)
-	}
+	password := "1234"
```
Comment (b00f) [inline cmd/daemon/start.go:189]: please revert this file

**Review by jwmdev — COMMENTED** [review]
Because of dependance issues of zmq on window, consider replacing it with nanomsg, preferably the one implemented using pure go to avoid any dependance issues: The repository is [nanomsg mango](https://github.com/nanomsg/mangos)

**Comment by b00f** [comment]
@jwmdev nanomsg looks a good option for us. Thanks for bringing it up.

**Comment by jwmdev** [comment]
Should we close this pull request another one one (#355 ) is requested?

**Comment by b00f** [comment]
Yes. We can close it now

**Pull request closed without merging**